openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -516,43 +519,3 @@ def is_enabled(self, course_key=None):  # pylint: disable=arguments-differ
             self.flag_name,
             check_before_waffle_callback=self._get_course_override_callback(course_key),
         )
-
-
-def _get_instance_module_name(instance_object):
-    """
-    Returns the module in which the passed instance object was defined.
-
-    This must be called from the ``__init__`` method of the instance class.
-    """
-    try:
-        stack = traceback.extract_stack()
-        # Example: 'openedx.core.djangoapps.waffle_utils'
-        instance_class_module = instance_object.__class__.__module__
-        # Example: 'openedx/core/djangoapps/waffle_utils'
-        class_module_as_path = instance_class_module.replace('.', '/')
-        module_string_index = -1
-        # start searching at -2, because -1 starts in this function.
-        # search up the stack to max of -10, because the class hierarchy
-        # shouldn't be that deep.
-        for stack_index in range(-2, -10, -1):
-            # Example: '/edx/app/edxapp/edx-platform/openedx/core/djangoapps/waffle_utils/__init__.py'
-            class_file_name = stack[stack_index].filename
-            if module_string_index == -1:
-                module_string_index = class_file_name.find(class_module_as_path)
-            if stack[stack_index].name != '__init__':
-                # assumes the instance is being defined as soon as we are
-                # no longer in any __init__ method of the class hierarchy.
-                break
-        # Example: '/edx/app/edxapp/edx-platform/openedx/core/djangoapps/programs/__init__.py'
-        instance_file_name = stack[stack_index].filename
-        # Example: 'openedx/core/djangoapps/programs/__init__.py'
-        instance_module_name = instance_file_name[module_string_index:]
-        # Example: 'openedx.core.djangoapps.programs.__init__.py'
-        instance_module_name = instance_module_name.replace('/', '.')
-        # Example: 'openedx.core.djangoapps.programs.__init__'
-        instance_module_name = re.sub(r'\.py$', '', instance_module_name)
-        # Example: 'openedx.core.djangoapps.programs'
-        instance_module_name = re.sub(r'\.__init__$', '', instance_module_name)
-        return instance_module_name
-    except Exception:  # pylint: disable=broad-except
-        return 'error.parsing.module'
```

Comment:
I'll miss you, hack.  You were fun to write. :)

Reply:
:yum: